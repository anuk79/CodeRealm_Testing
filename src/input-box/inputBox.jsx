
import React, { Component } from 'react';

export class InputBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value || '',
      error: ''
    }
  }

  setvalue = (value) => {
    this.setState({
      value: value,
      error: !value
    })
  }

  render() {
    return (
      <div>
        <label>Enter your name</label>
        <input
          type="text"
          value={this.state.value}
          onChange={(event) => this.setvalue(event.target.value)} />
        <button disabled={this.state.error}> Submit </button>
        {this.state.error && <div className='error-message'>The value entered is not valid.</div>}
        <div>{this}</div>
      </div>
    );
  }
} 