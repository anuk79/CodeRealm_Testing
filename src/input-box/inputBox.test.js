
// vendors
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// components
import { InputBox } from './inputBox';

describe.skip('InputBox component', () => {
  let wrapper;

  beforeEach(()=> {
    wrapper = shallow(<InputBox />);
  });
  
  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('input').length).toBe(1);
  });

  it('should set state on blur event of input', () => {
    const setStateSpy = jest.spyOn(wrapper.instance(), 'setState');

    wrapper.find('input').simulate('change', { target: { value: 123 } });
    expect(setStateSpy).toBeCalled();
  });
});





















// jest.mock('react-router-dom', () => ({
//   withRouter: component => component,
//   Link: 'Link'
// }));
