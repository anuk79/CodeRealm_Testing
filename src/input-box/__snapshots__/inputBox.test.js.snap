// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox component should render correctly 1`] = `
<div>
  <label>
    Enter your name
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    disabled=""
  >
     Submit 
  </button>
  <div>
    <Component />
  </div>
</div>
`;
